From ab5ca07af03b7dbb33193666c13b938534bde0e4 Mon Sep 17 00:00:00 2001
From: Sarath Lakshman <sarath@couchbase.com>
Date: Sat, 11 Mar 2023 16:58:31 +0530
Subject: [PATCH] Fix max_threads command line parameter propagation

The fuse_main_real() method doesn't apply the max_threads parameter
parsed through the commandline arguments. This commit fixes the wiring
of max_threads argument.
---
 lib/helper.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/helper.c b/lib/helper.c
index 35c6a98c..14a0df33 100644
--- a/lib/helper.c
+++ b/lib/helper.c
@@ -377,6 +377,7 @@ int fuse_main_real(int argc, char *argv[], const struct fuse_operations *op,
 		fuse_loop_cfg_set_clone_fd(loop_config, opts.clone_fd);
 
 		fuse_loop_cfg_set_idle_threads(loop_config, opts.max_idle_threads);
+		fuse_loop_cfg_set_max_threads(loop_config, opts.max_threads);
 		res = fuse_loop_mt(fuse, loop_config);
 	}
 	if (res)
